### PR TITLE
Ports a Modified Version of the CM Defib Charge Sound

### PR DESCRIPTION
**Pull request by Fordoxia**
## What Does This PR Do
Changes the defib's charging sound. It now plays the moment you start placing the paddles on the patient, giving an immediate audible warning to anyone nearby.

Preview: 

https://github.com/ParadiseSS13/Paradise/assets/143041327/2cb596e5-0cca-4d32-912e-7e35b42a8c8c

## Why It's Good For The Game
The current defib only gives you 2 seconds of warning before applying a shock. This is not much time to react to something that can easily kill you with a heart attack if you are administering CPR.

Defibs now give a full 5 seconds of warning and have a much more distinctive sound that tells you to stop prodding the patient for a moment. This will help reduce accidental fatalities in medical.

Also it tells people to STOP PUSHING YOU!

## Testing
Beat the crap out of a skrell. Defib'd a skrell. Tried to defib myself.

## Changelog
:cl:
tweak: There is now a single do_after lasting 5 seconds when defibbing, rather than two seperate 3 and 2 second ones.
soundadd: New sound for the defib.
/:cl:

### CODE REVIEW

**Review by Spaghetti-bit — CHANGES_REQUESTED** [review]
code is approved, please provide an example of the sounds in a video.

**Comment by Fordoxia** [comment]
> code is approved, please provide an example of the sounds in a video.

Done.

File: code/datums/components/defibrillator.dm [hunk 1]
```diff
@@ -24,7 +24,20 @@
 	var/emag_proof
 	/// uid to an item that should be making noise and handling things that our direct parent shouldn't be concerned with.
 	var/actual_unit_uid
-
+	/// The time it takes between clicking on the patient and the zap being delivered.
+	var/defib_time = 5 SECONDS
```
Comment (Contrabang) [inline code/datums/components/defibrillator.dm:28]: this should probably be a define, so that admins dont mess with it

**Review by Contrabang — DISMISSED** [review]

File: code/datums/components/defibrillator.dm (hunk 1 above)
Comment (Fordoxia) [reply code/datums/components/defibrillator.dm:28]: T R U E

**Review by Spaghetti-bit — APPROVED** [review]
approved for code and for witnessing a man explode in the demonstration video.

**Review by Sirryan2002 — APPROVED** [review]

File: code/datums/components/defibrillator.dm [hunk 2]
```diff
@@ -78,11 +93,11 @@
 	if(safety)
 		safety = FALSE
 		unit.visible_message("<span class='notice'>[unit] beeps: Safety protocols disabled!</span>")
-		playsound(get_turf(unit), 'sound/machines/defib_saftyoff.ogg', 50, 0)
+		playsound(get_turf(unit), safety_off_sound, 50, 0)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:96]:
```suggestion
		playsound(get_turf(unit), safety_off_sound, 50, FALSE)
```

File: code/datums/components/defibrillator.dm [hunk 3]
```diff
@@ -78,11 +93,11 @@
...
 	else
 		safety = TRUE
 		unit.visible_message("<span class='notice'>[unit] beeps: Safety protocols enabled!</span>")
-		playsound(get_turf(unit), 'sound/machines/defib_saftyon.ogg', 50, 0)
+		playsound(get_turf(unit), safety_on_sound, 50, 0)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:100]:
```suggestion
		playsound(get_turf(unit), safety_on_sound, 50, FALSE)
```

File: code/datums/components/defibrillator.dm [hunk 4]
```diff
@@ -134,7 +149,7 @@
 
 	if(application_result & COMPONENT_BLOCK_DEFIB_DEAD)
 		user.visible_message("<span class='notice'>[defib_ref] beeps: Unit is unpowered.</span>")
-		playsound(get_turf(defib_ref), 'sound/machines/defib_failed.ogg', 50, 0)
+		playsound(get_turf(defib_ref), fail_sound, 50, 0)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:152]:
```suggestion
		playsound(get_turf(defib_ref), fail_sound, 50, FALSE)
```

File: code/datums/components/defibrillator.dm [hunk 5]
```diff
@@ -179,26 +194,22 @@
 		SEND_SOUND(target, sound('sound/effects/genetics.ogg'))
 		target.throw_alert("cling_defib", /atom/movable/screen/alert/changeling_defib_revive, alert_args = list(parent, target))
 
-	if(!do_after(user, 3 SECONDS * speed_multiplier, target = target)) // Beginning to place the paddles on patient's chest to allow some time for people to move away to stop the process
-		busy = FALSE
-		return
-
 	user.visible_message("<span class='notice'>[user] places [parent] on [target]'s chest.</span>", "<span class='warning'>You place [parent] on [target]'s chest.</span>")
-	playsound(get_turf(defib_ref), 'sound/machines/defib_charge.ogg', 50, 0)
+	playsound(get_turf(defib_ref), charge_sound, 50, 0)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:198]:
```suggestion
	playsound(get_turf(defib_ref), charge_sound, 50, FALSE)
```

File: code/datums/components/defibrillator.dm [hunk 6]
```diff
@@ -179,26 +194,22 @@
...
 
-	if(!do_after(user, 2 SECONDS * speed_multiplier, target = target)) // Placed on chest and short delay to shock for dramatic effect, revive time is 5sec total
+	if(!do_after(user, DEFIB_TIME * speed_multiplier, target = target)) // Placed on chest and short delay to shock for dramatic effect, revive time is 5sec total
 		busy = FALSE
 		return
 
 	if(istype(target.wear_suit, /obj/item/clothing/suit/space) && !combat)
 		user.visible_message("<span class='notice'>[defib_ref] buzzes: Patient's chest is obscured. Operation aborted.</span>")
-		playsound(get_turf(defib_ref), 'sound/machines/defib_failed.ogg', 50, 0)
+		playsound(get_turf(defib_ref), fail_sound, 50, 0)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:212]:
```suggestion
		playsound(get_turf(defib_ref), fail_sound, 50, FALSE)
```

File: code/datums/components/defibrillator.dm [hunk 7]
```diff
@@ -216,7 +227,7 @@
 		else if(heart.linked_organ.status & ORGAN_DEAD)
 			user.visible_message("<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Heart necrosis detected.</span>")
 		if(!heart || (heart.linked_organ.status & ORGAN_DEAD))
-			playsound(get_turf(defib_ref), 'sound/machines/defib_failed.ogg', 50, 0)
+			playsound(get_turf(defib_ref), fail_sound, 50, 0)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:230]:
```suggestion
			playsound(get_turf(defib_ref), fail_sound, 50, FALSE)
```

File: code/datums/components/defibrillator.dm [hunk 8]
```diff
@@ -226,21 +237,21 @@
 		set_cooldown(cooldown)
 		user.visible_message("<span class='boldnotice'>[defib_ref] pings: Cardiac arrhythmia corrected.</span>")
 		target.visible_message("<span class='warning'>[target]'s body convulses a bit.</span>", "<span class='userdanger'>You feel a jolt, and your heartbeat seems to steady.</span>")
-		playsound(get_turf(defib_ref), 'sound/machines/defib_zap.ogg', 50, 1, -1)
+		playsound(get_turf(defib_ref), zap_sound, 50, 1, -1)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:240]:
```suggestion
		playsound(get_turf(defib_ref), zap_sound, 50, TRUE, -1)
```

File: code/datums/components/defibrillator.dm [hunk 9]
```diff
@@ -226,21 +237,21 @@
...
 		playsound(get_turf(defib_ref), "bodyfall", 50, 1)
-		playsound(get_turf(defib_ref), 'sound/machines/defib_success.ogg', 50, 0)
+		playsound(get_turf(defib_ref), success_sound, 50, 0)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:242]:
```suggestion
		playsound(get_turf(defib_ref), success_sound, 50, FALSE)
```

File: code/datums/components/defibrillator.dm [hunk 10]
```diff
@@ -226,21 +237,21 @@
 		set_cooldown(cooldown)
 		user.visible_message("<span class='boldnotice'>[defib_ref] pings: Cardiac arrhythmia corrected.</span>")
 		target.visible_message("<span class='warning'>[target]'s body convulses a bit.</span>", "<span class='userdanger'>You feel a jolt, and your heartbeat seems to steady.</span>")
-		playsound(get_turf(defib_ref), 'sound/machines/defib_zap.ogg', 50, 1, -1)
+		playsound(get_turf(defib_ref), zap_sound, 50, 1, -1)
 		playsound(get_turf(defib_ref), "bodyfall", 50, 1)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:241]:
```suggestion
		playsound(get_turf(defib_ref), "bodyfall", 50, TRUE)
```
Might as well fix this one too.

File: code/datums/components/defibrillator.dm [hunk 11]
```diff
@@ -226,21 +237,21 @@
...
-		playsound(get_turf(defib_ref), 'sound/machines/defib_success.ogg', 50, 0)
+		playsound(get_turf(defib_ref), success_sound, 50, 0)
 		busy = FALSE
 		return
 
 	if(target.stat != DEAD && !HAS_TRAIT(target, TRAIT_FAKEDEATH))
 		user.visible_message("<span class='notice'>[defib_ref] buzzes: Patient is not in a valid state. Operation aborted.</span>")
-		playsound(get_turf(defib_ref), 'sound/machines/defib_failed.ogg', 50, 0)
+		playsound(get_turf(defib_ref), fail_sound, 50, 0)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:248]:
```suggestion
		playsound(get_turf(defib_ref), fail_sound, 50, FALSE)
```

File: code/datums/components/defibrillator.dm [hunk 12]
```diff
@@ -226,21 +237,21 @@
...
 		busy = FALSE
 		return
 
 	target.visible_message("<span class='warning'>[target]'s body convulses a bit.</span>")
 	playsound(get_turf(defib_ref), "bodyfall", 50, 1)
-	playsound(get_turf(defib_ref), 'sound/machines/defib_zap.ogg', 50, 1, -1)
+	playsound(get_turf(defib_ref), zap_sound, 50, 1, -1)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:254]:
```suggestion
	playsound(get_turf(defib_ref), zap_sound, 50, TRUE, -1)
```

File: code/datums/components/defibrillator.dm [hunk 13]
```diff
@@ -226,21 +237,21 @@
...
 
 	if(target.stat != DEAD && !HAS_TRAIT(target, TRAIT_FAKEDEATH))
 		user.visible_message("<span class='notice'>[defib_ref] buzzes: Patient is not in a valid state. Operation aborted.</span>")
-		playsound(get_turf(defib_ref), 'sound/machines/defib_failed.ogg', 50, 0)
+		playsound(get_turf(defib_ref), fail_sound, 50, 0)
 		busy = FALSE
 		return
 
 	target.visible_message("<span class='warning'>[target]'s body convulses a bit.</span>")
 	playsound(get_turf(defib_ref), "bodyfall", 50, 1)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:253]:
```suggestion
	playsound(get_turf(defib_ref), "bodyfall", 50, TRUE)
```
This one as well.

File: code/datums/components/defibrillator.dm [hunk 14]
```diff
@@ -276,7 +287,7 @@
 
 	if(failure_message)
 		user.visible_message(failure_message)
-		playsound(get_turf(defib_ref), 'sound/machines/defib_failed.ogg', 50, 0)
+		playsound(get_turf(defib_ref), fail_sound, 50, 0)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:290]:
```suggestion
		playsound(get_turf(defib_ref), fail_sound, 50, FALSE)
```

File: code/datums/components/defibrillator.dm [hunk 15]
```diff
@@ -299,10 +310,10 @@
 
 		if(target.getBrainLoss() >= 100)
 			// If you want to treat this with mannitol, it'll have to metabolize while the patient is alive, so it's alright to bring them back up for a minute
-			playsound(get_turf(defib_ref), 'sound/machines/defib_saftyoff.ogg', 50, 0)
+			playsound(get_turf(defib_ref), safety_off_sound, 50, 0)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:313]:
```suggestion
			playsound(get_turf(defib_ref), safety_off_sound, 50, FALSE)
```

File: code/datums/components/defibrillator.dm [hunk 16]
```diff
@@ -299,10 +310,10 @@
...
 			user.visible_message("<span class='boldnotice'>[defib_ref] chimes: Minimal brain activity detected, brain treatment recommended for full resuscitation.</span>")
 		else
-			playsound(get_turf(defib_ref), 'sound/machines/defib_success.ogg', 50, 0)
+			playsound(get_turf(defib_ref), success_sound, 50, 0)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:316]:
```suggestion
			playsound(get_turf(defib_ref), success_sound, 50, FALSE)
```

File: code/datums/components/defibrillator.dm [hunk 17]
```diff
@@ -340,7 +351,7 @@
 			"<span class='userdanger'>[user] touches you with [parent], and you feel a strong jolt!</span>")
 	target.apply_damage(60, STAMINA)
 	target.KnockDown(10 SECONDS)
-	playsound(get_turf(parent), 'sound/machines/defib_zap.ogg', 50, 1, -1)
+	playsound(get_turf(parent), zap_sound, 50, 1, -1)
```
Comment (Burzah) [inline code/datums/components/defibrillator.dm:354]:
```suggestion
	playsound(get_turf(parent), zap_sound, 50, TRUE, -1)
```

**Review by Burzah — CHANGES_REQUESTED** [review]

**Review by Burzah — APPROVED** [review]

**Comment by Sirryan2002** [comment]
@Contrabang

**Comment by Sirryan2002** [comment]
Dawg you need to fix your master branch

**Comment by Sirryan2002** [comment]
@Contrabang

**Review by S34NW — APPROVED** [review]

**Pull request merged**